src/device.rs

```
@@ -1642,6 +1653,22 @@ impl Runtime {
             .await?;
         Ok(())
     }
+    async fn set_tun(&mut self, tun: i32) -> Result {
+        self.requested_state.device_config.name = None;
+        #[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos"))]
```

Should'nt android be in this list as well?

Also maybe it would make sense to check if neptun is our current WG adapter of choice and return something like "unsupported" in other cases in order to minimize silent failures?

---

Yup, it should - fixed.

The check for adapter doesn't need happen here - the error will be propagated from the non NepTUN adapter in this call:

```
self.entities.wireguard_interface.set_tun(tun).await?;
```